### PR TITLE
Fix react merged types

### PR DESCRIPTION
**Pull request by pedrodurek**
Closes https://github.com/i18next/react-i18next/issues/1543

#### Checklist

- [x] only relevant code is changed (make a diff before you submit the PR)
- [x] run tests `npm run test`
- [ ] tests are included
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/i18next/.github/blob/master/CONTRIBUTING.md)

#### Checklist (for documentation change)

- [x] only relevant documentation part is changed (make a diff before you submit the PR)
- [x] motivation/reason is provided
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/i18next/.github/blob/master/CONTRIBUTING.md)

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -39,11 +39,10 @@ declare module 'i18next' {
 type ObjectOrNever = TypeOptions['allowObjectInHTMLChildren'] extends true
   ? Record<string, unknown>
   : never;
-type ReactI18NextChild = React.ReactNode | ObjectOrNever;
 
 declare module 'react' {
   interface HTMLAttributes<T> {
-    children?: ReactI18NextChild | Iterable<ReactI18NextChild>;
```
Comment (pedrodurek) [inline index.d.ts:46]: I don't remember why and when we did that, but it doesn't make much sense, since `ReactNode` is already `Iterable`

**Comment by adrai** [comment]
patch version ok?

btw:
image

File: index.d.ts [hunk 2]
```diff
@@ -39,11 +39,10 @@ declare module 'i18next' {
...
+    children?: React.ReactNode | ObjectOrNever;
```
Comment (pedrodurek) [inline index.d.ts:45]: Now, children will behave exactly like the [actual](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/react/v17/index.d.ts#L1342) unless `allowObjectInHTMLChildren` is `true`.

File: index.d.ts (hunk 1 above)
Comment (adrai) [reply index.d.ts]: originally introduced with #1531 by @gshokanov

Comment (adrai) [reply index.d.ts]: //cc @lexanth @cravend @Igosuki @gabriel-dehan

File: index.d.ts [hunk 3]
```diff
@@ -39,11 +39,12 @@ declare module 'i18next' {
 type ObjectOrNever = TypeOptions['allowObjectInHTMLChildren'] extends true
   ? Record<string, unknown>
   : never;
-type ReactI18NextChild = React.ReactNode | ObjectOrNever;
+
+type ReactI18NextChildren = React.ReactNode | ObjectOrNever;
```
Comment (pedrodurek) [inline index.d.ts:43]: Now, children will behave exactly like the [actual](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/react/v17/index.d.ts#L1342) unless `allowObjectInHTMLChildren` is `true`.

**Comment by pedrodurek** [comment]
@adrai, yes patch version

**Comment by adrai** [comment]
included in v12.1.5

**Comment by YutaMoriJP** [comment]
Was this change overridden in the latest release (13.2.0)?
> This JSX tag's 'children' prop expects a single child of type 'ReactI18NextChildren', but multiple children were provided.

**Comment by adrai** [comment]
@YutaMoriJP don't think so... correct @nicegamer7 ?

**Comment by nicegamer7** [comment]
It shouldn't have been affected, no.

Though, at the company I'm working at, we had issues with this option already. I'll create a PR with the fix I implemented. I didn't open this PR previously because I didn't see anyone else complaining about this issue.

**Comment by YutaMoriJP** [comment]
@nicegamer7 Oh gosh yes please 🙏
Maybe before you create the PR mind telling me how you fixed it?  Kinda stuck with this one... Thanks!

**Comment by YutaMoriJP** [comment]
@nicegamer7 Never mind you already created PR and it got merged. You're fast! Thank you!

**Pull request merged**